internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -458,25 +420,38 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			err = errors.Wrap(err, errRemoveFinalizer)
 			record.Event(cm, event.Warning(reasonDelete, err))
 			cm.SetConditions(xpv1.ReconcileError(err))
-			return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
+			return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, cm, r.fieldOwner), errUpdateClaimStatus)
 		}
 
 		log.Debug("Successfully deleted composite resource claim")
 		cm.SetConditions(xpv1.ReconcileSuccess())
-		return reconcile.Result{Requeue: false}, errors.Wrap(r.client.Status().Update(ctx, cm), errUpdateClaimStatus)
+		return reconcile.Result{Requeue: false}, errors.Wrap(r.client.Status().Update(ctx, cm, r.fieldOwner), errUpdateClaimStatus)
 	}
 
-	if err := r.claim.AddFinalizer(ctx, cm); err != nil {
-		if kerrors.IsConflict(err) {
-			return reconcile.Result{Requeue: true}, nil
+	// If composite exists, i.e. got created in the previous iteration,
+	// it might be needed to fix field ownerships,
+	// so that the claim controller becomes an exclusive owner
+	// of the fields mirrored from the claim to the composite
+	if meta.WasCreated(cp) && r.maybeFixFieldOwnership(cp) {
+		if err := r.client.Update(ctx, cp); err != nil {
```

Comment:
Does this mean that annotations, labels, finalizers are now all owned by the controller? That sounds wrong and dangerous.

Reply:
How can the update know which fields owned by other managers should be taken over?

Reply:
Oh, got it. Please move `r.maybeFixFieldOwnership(cp)` into its own if clause to make it clear that it has side-effects.

Reply:
done.